rasa/shared/nlu/training_data/training_data.py

```diff
@@ -679,38 +678,54 @@ def can_train_nlu_model(self) -> bool:
         ]
         return not any([len(lst) > 0 for lst in lists_to_check])
 
-    def divide_into_chunks(self, max_size: int = 2) -> List["TrainingDataChunk"]:
+    def divide_into_chunks(self, num_chunks: int) -> List["TrainingDataChunk"]:
         """Divides the training data into smaller chunks.
 
         Each chunk should be a good representation of the complete dataset. E.g. it
         should not just contain examples of one intent, but instead match the
         distribution of the complete dataset.
 
         Args:
-            max_size: The maximum size (in MB) of one chunk.
+            num_chunks: The total number of chunks into which the training data should be broken.
 
         Returns:
             A list of all training data chunks.
         """
-        pass
+        all_chunks = []
+        data_to_chunk = copy.deepcopy(self)
```

Comment:
deepcopy doubles the memory as far as I know. Can we somehow avoid it?

Reply:
Removed the copy. `split_nlu_examples` doesn't modify the original data and hence we don't need the copy anymore.